File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/python/F17.P2.py
released = 0


# BEGIN PROBE F17.P2
class Handle:
    def __init__(self, id):
        self.id = id

    def __enter__(self):
        return self

    def __exit__(self, *exc):
        global released
        released += 1


with Handle(1) as h:
    pass

result = released
# END PROBE F17.P2


print(result)

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F17.P2.ts
let released = 0;

// BEGIN PROBE F17.P2
class Handle {
  constructor(readonly id: number) {}
  close() { released++; }
}
{
  const h = new Handle(1);
  try { } finally { h.close(); }
}
const result = released;
// END PROBE F17.P2

console.log(result);
